QGL/Cliffords.py

```diff
@@ -12,103 +12,116 @@
+pY = np.array([[0, -1j], [1j, 0]], dtype=np.complex128)
+pZ = np.array([[1, 0], [0, -1]], dtype=np.complex128)
 pI = np.eye(2, dtype=np.complex128)
 
+
 def pauli_mats(n):
-	"""
+    """
 	Return a list of n-qubit Paulis as numpy array.
 	"""
-	assert n > 0, "You need at least 1 qubit!"
-	if n == 1:
-		return [pI, pX, pY, pZ]
-	else:
-		paulis = pauli_mats(n-1)
-		return [np.kron(p1, p2) for p1,p2 in product([pI, pX, pY, pZ], paulis)]
+    assert n > 0, "You need at least 1 qubit!"
+    if n == 1:
+        return [pI, pX, pY, pZ]
+    else:
+        paulis = pauli_mats(n - 1)
+        return [np.kron(p1, p2)
+                for p1, p2 in product([pI, pX, pY, pZ], paulis)]
 
 #Basis single-qubit Cliffords with an arbitrary enumeration
 C1 = {}
 C1[0] = pI
-C1[1] = expm(-1j*(pi/4)*pX)
-C1[2] = expm(-2j*(pi/4)*pX)
-C1[3] = expm(-3j*(pi/4)*pX)
-C1[4] = expm(-1j*(pi/4)*pY)
-C1[5] = expm(-2j*(pi/4)*pY)
-C1[6] = expm(-3j*(pi/4)*pY)
-C1[7] = expm(-1j*(pi/4)*pZ)
-C1[8] = expm(-2j*(pi/4)*pZ)
-C1[9] = expm(-3j*(pi/4)*pZ)
-C1[10] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pX+pY))
-C1[11] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pX-pY))
-C1[12] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pX+pZ))
-C1[13] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pX-pZ))
-C1[14] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pY+pZ))
-C1[15] = expm(-1j*(pi/2)*(1/np.sqrt(2))*(pY-pZ))
-C1[16] = expm(-1j*(pi/3)*(1/np.sqrt(3))*(pX+pY+pZ))
-C1[17] = expm(-2j*(pi/3)*(1/np.sqrt(3))*(pX+pY+pZ))
-C1[18] = expm(-1j*(pi/3)*(1/np.sqrt(3))*(pX-pY+pZ))
-C1[19] = expm(-2j*(pi/3)*(1/np.sqrt(3))*(pX-pY+pZ))
-C1[20] = expm(-1j*(pi/3)*(1/np.sqrt(3))*(pX+pY-pZ))
-C1[21] = expm(-2j*(pi/3)*(1/np.sqrt(3))*(pX+pY-pZ))
-C1[22] = expm(-1j*(pi/3)*(1/np.sqrt(3))*(-pX+pY+pZ))
-C1[23] = expm(-2j*(pi/3)*(1/np.sqrt(3))*(-pX+pY+pZ))
+C1[1] = expm(-1j * (pi / 4) * pX)
+C1[2] = expm(-2j * (pi / 4) * pX)
+C1[3] = expm(-3j * (pi / 4) * pX)
+C1[4] = expm(-1j * (pi / 4) * pY)
+C1[5] = expm(-2j * (pi / 4) * pY)
+C1[6] = expm(-3j * (pi / 4) * pY)
+C1[7] = expm(-1j * (pi / 4) * pZ)
+C1[8] = expm(-2j * (pi / 4) * pZ)
+C1[9] = expm(-3j * (pi / 4) * pZ)
+C1[10] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pX + pY))
+C1[11] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pX - pY))
+C1[12] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pX + pZ))
+C1[13] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pX - pZ))
+C1[14] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pY + pZ))
+C1[15] = expm(-1j * (pi / 2) * (1 / np.sqrt(2)) * (pY - pZ))
+C1[16] = expm(-1j * (pi / 3) * (1 / np.sqrt(3)) * (pX + pY + pZ))
+C1[17] = expm(-2j * (pi / 3) * (1 / np.sqrt(3)) * (pX + pY + pZ))
+C1[18] = expm(-1j * (pi / 3) * (1 / np.sqrt(3)) * (pX - pY + pZ))
+C1[19] = expm(-2j * (pi / 3) * (1 / np.sqrt(3)) * (pX - pY + pZ))
+C1[20] = expm(-1j * (pi / 3) * (1 / np.sqrt(3)) * (pX + pY - pZ))
+C1[21] = expm(-2j * (pi / 3) * (1 / np.sqrt(3)) * (pX + pY - pZ))
+C1[22] = expm(-1j * (pi / 3) * (1 / np.sqrt(3)) * (-pX + pY + pZ))
+C1[23] = expm(-2j * (pi / 3) * (1 / np.sqrt(3)) * (-pX + pY + pZ))
```

Comment:
It seems like it would be nice for there to be a setting that if the numerator or denominator of a division is a constant, we'd like to skip the space. Alternatively, just disable yapf for this entire block.